backend/test/api/index.test.js

```diff
@@ -331,3 +335,89 @@ test.serial('POST /api/resubscribe return error when already subscribed', async
 
 	jobsToDelete.forEach(job => job.remove());
 });
+
+test('POST /api/delete-account returns 401 - without token', async t => {
+	const response = await request(app).post('/api/delete-account');
+	t.is(response.status, UNAUTHORIZED);
+	t.is(response.body.errors.message, 'User token is invalid');
+});
+
+test.serial('POST /api/delete-account schedules user deletion job - with appropriate token', async t => {
+	const { id, email, username } = t.context.user;
+	const user = { id, email };
+	const token = generateToken(user);
+
+	const scheduleDeletionOfUser = sinon.stub(UserScheduler.prototype, 'scheduleDeletionOfUser');
+	scheduleDeletionOfUser.returns();
+
+	const response = await request(app)
+		.post('/api/delete-account')
+		.set('authorization', JSON.stringify({ token }));
+
+	t.is(response.status, OK);
+	t.is(response.body.message, `Successfully scheduled deletion of user ${username}`);
+
+	t.true(scheduleDeletionOfUser.calledOnce);
+	scheduleDeletionOfUser.restore();
+});
+
+test.serial('POST /api/delete-account returns error - when user does not exist', async t => {
+	const user = { id: 1337, email: 'foo@bar.com' };
+	const token = generateToken(user);
+
+	const response = await request(app)
+		.post('/api/delete-account')
+		.set('authorization', JSON.stringify({ token }));
+
+	t.is(response.status, NOT_FOUND);
+	t.is(response.body.errors.message, 'The user that you are trying to delete does not exist');
+});
+
+test('POST /api/cancel-deletion returns 401 - without token', async t => {
+	const response = await request(app).post('/api/cancel-deletion');
+	t.is(response.status, UNAUTHORIZED);
+	t.is(response.body.errors.message, 'User token is invalid');
+});
+
+test.serial('POST /api/cancel-deletion removes deleteAccount job - with appropriate token', async t => {
+	const { id, email, username } = t.context.user;
+	const user = { id, email };
+	const token = generateToken(user);
+
+	await deleteAccountJob(user);
```

Comment:
Consider renaming to `createAccountDeletionJob`

Reply:
Will rename all methods that generate jobs as recommended, after this PR has been merged.